service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpConnectionHandler.java

```diff
@@ -213,4 +277,10 @@ private void addProxyLayer(Transport transport, String hostName)
         proxy.configure(hostName + ":" + AmqpConnectionHandler.AMQPS_WS_PORT, null, proxyHandler, transport);
         ((TransportInternal) transport).addTransportLayer(proxy);
     }
+
+    @Override
+    public void onAuthenticationFailed(Exception e)
+    {
+        this.savedException = new IOException("Failed to authenticate connection", e);
```

Comment:
Throw IotHubException instead?  This does not look like IOException.

Reply:
Good idea. We only hit this when the communication between hub and client went fine, just that hub throws an unauthorized exception or something similar.